lib/Doctrine/ORM/EntityManager.php

```diff
@@ -835,24 +836,33 @@ public static function create($conn, Configuration $config, EventManager $eventM
             throw ORMException::missingMappingDriverImpl();
         }
 
-        switch (true) {
-            case (is_array($conn)):
-                $conn = \Doctrine\DBAL\DriverManager::getConnection(
-                    $conn, $config, ($eventManager ?: new EventManager())
-                );
-                break;
+        $conn = static::createConnection($conn, $config, $eventManager);
 
-            case ($conn instanceof Connection):
-                if ($eventManager !== null && $conn->getEventManager() !== $eventManager) {
-                     throw ORMException::mismatchedEventManager();
-                }
-                break;
+        return new EntityManager($conn, $config, $conn->getEventManager());
+    }
 
-            default:
-                throw new \InvalidArgumentException("Invalid argument: " . $conn);
+    /**
+     * Factory method to create Connection instances.
+     *
+     * @param array|Connection $conn         An array with the connection parameters or an existing Connection instance.
+     * @param Configuration    $config       The Configuration instance to use.
+     * @param EventManager     $eventManager The EventManager instance to use.
+     *
+     * @return Connection
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected static function createConnection($conn, Configuration $config, EventManager $eventManager = null)
+    {
+        if (is_array($conn)) {
+            return DriverManager::getConnection(
+                $conn, $config, ($eventManager ?: new EventManager())
```

Comment:
This syntax is not PSR-2-compliant: split over multiple rows or pack it into a single row